lib/swm050/wdt.c

```diff
@@ -0,0 +1,158 @@
+/** @defgroup wdg_file Watchdog peripheral API
+ * @brief SWM050 WDT API.
+ * @ingroup peripheral_apis
+ * LGPL License Terms @ref lgpl_license
+ * @author @htmlonly &copy; @endhtmlonly 2019
+ * Caleb Szalacinski <contact@skiboy.net>
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2019 Caleb Szalacinski <contact@skiboy.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**@{*/
+#include <libopencm3/swm050/wdt.h>
+#include <libopencm3/swm050/sysctl.h>
+/*---------------------------------------------------------------------------*/
+/** Enables the WDT.
+
+Enables the WDT.  Before calling this, some setup needs to be done.
+It should look like this:
+
+	wdt_clock_enable(1);
+	wdt_set_time(time1, time2);
+	wdt_mode(mode);
+	wdt_restart();
+
+If WDT_MODE_INT is used, the interrupt should also be enabled
+using the NVIC.
+
+@param en True to enable, false to disable
```

Comment:
you could just make the parameter "enable" here, and not have to write the documentation for it.  It's not a parameter the user has to type, so there's no reason to scrimp on characters :)